### PR TITLE
Updated tc.plugin (linux bandwidth QoS)

### PR DESCRIPTION
**Pull request by ktsaou**
- [x] replaced AVL with DICTIONARY
- [x] replaced traversal linked lists with DICTIONARY
- [x] replaced `char *` with `STRING *`
- [x] replaced custom TC monitoring charts with worker utilization custom metrics
- [x] used DICTIONARY for constructor/destructor of network devices and classes/qdiscs
- [x] added labels to allow slicing and dicing the data under netdata cloud dashboards

### CODE REVIEW

**Review by thiagoftsm — APPROVED** [review]
I compiled this PR with success on `Slackware current` and `Ubuntu 18.04`, both on x86_64. I also compiled on `Armv7l` and I did not have any warning related to this PR.

Finally I tested it on `Ubuntu 18.04` and I got the expected result:

Screenshot_20220906_225444

The most important logs are not showing nothing to have concerns:

```sh
2022-09-06 22:44:49: netdata INFO  : PLUGIN[tc] : (0137@libnetdata/threads/t:thread_set_name): set name of thread 1015 to PLUGIN[tc]
2022-09-06 22:44:50: netdata INFO  : PLUGIN[tc] : (0184@database/sqlite/sqli:prepare_stateme): Thread 1015: Using key 1 on statement 0x55903e3b5ab8
2022-09-06 22:44:50: netdata INFO  : PLUGIN[tc] : (0184@database/sqlite/sqli:prepare_stateme): Thread 1015: Using key 2 on statement 0x55903e3b7378
2022-09-06 22:44:50: netdata INFO  : PLUGIN[tc] : (0184@database/sqlite/sqli:prepare_stateme): Thread 1015: Using key 3 on statement 0x55903e3c1538
2022-09-06 22:44:50: netdata INFO  : PLUGIN[tc] : (0184@database/sqlite/sqli:prepare_stateme): Thread 1015: Using key 4 on statement 0x55903e3c1bc8
2022-09-06 22:44:50: netdata IERR  : PLUGIN[tc] : (2708@database/rrdcontext.:rrdcontext_queu): RRDCONTEXT: 'tc.qos' update triggered by function rrdcontext_react_callback(), due to flags: UPDATED , reasons: object created
2022-09-06 22:44:50: netdata INFO  : PLUGIN[tc] : (0184@database/sqlite/sqli:prepare_stateme): Thread 1015: Using key 5 on statement 0x55903e3c2198
2022-09-06 22:44:50: netdata INFO  : PLUGIN[tc] : (0184@database/sqlite/sqli:prepare_stateme): Thread 1015: Using key 6 on statement 0x55903e3c5368
2022-09-06 22:44:50: netdata INFO  : PLUGIN[tc] : (0184@database/sqlite/sqli:prepare_stateme): Thread 1015: Using key 7 on statement 0x55903e3c91b8
2022-09-06 22:44:50: netdata IERR  : PLUGIN[tc] : (2708@database/rrdcontext.:rrdcontext_queu): RRDCONTEXT: 'tc.qos_packets' update triggered by function rrdcontext_react_callback(), due to flags: UPDATED , reasons: object created
2022-09-06 22:44:50: netdata IERR  : PLUGIN[tc] : (2708@database/rrdcontext.:rrdcontext_queu): RRDCONTEXT: 'tc.qos_packets' update triggered by function rrdinstance_react_callback(), due to flags: UPDATED , reasons: object created
2022-09-06 22:44:51: netdata IERR  : PLUGIN[tc] : (2708@database/rrdcontext.:rrdcontext_queu): RRDCONTEXT: 'tc.qos' update triggered by function rrdmetric_collected_rrddim(), due to flags: COLLECTED UPDATED LIVE_RETENTION , reasons: sta
```
, LGTM!

File: collectors/tc.plugin/plugin_tc.c [hunk 1]
```diff
@@ -12,60 +12,48 @@
...
+    STRING *name;
+    STRING *leafid;
+    STRING *parentid;
+
+    bool hasparent:1;
+    bool isleaf:1;
+    bool isqdisc:1;
+    bool render:1;
+    bool name_updated:1;
+    bool updated:1;
```
Comment (vkalintiris) [inline collectors/tc.plugin/plugin_tc.c:25]: The only portable, non-implementation defined type that makes sense to use as a bitfield is a signed integer. Just use plain bools, here.

Comment (ktsaou) [reply collectors/tc.plugin/plugin_tc.c]: thanks! solved it.

File: collectors/tc.plugin/plugin_tc.c [hunk 2]
```diff
@@ -12,60 +12,48 @@
...
-    struct tc_class *next;
-    struct tc_class *prev;
 };
 
 struct tc_device {
-    avl_t avl;
-
-    char *id;
+    STRING *id;
     uint32_t hash;
```
Comment (vlvkobal) [inline collectors/tc.plugin/plugin_tc.c:50]: We don't use it anymore.

Comment (ktsaou) [reply collectors/tc.plugin/plugin_tc.c:50]: I am confused. What we don't use? `STRING *id` on `struct tc_class`?

Comment (ktsaou) [reply collectors/tc.plugin/plugin_tc.c:50]: Oh! `hash`! You are right!

**Review by vlvkobal — APPROVED** [review]
I haven't tested the plugin, but the code LGTM.

**Review by thiagoftsm — APPROVED** [review]
I retest the plugin with last commits and it is working, lgtm!

**Pull request merged**